### PR TITLE
CPLP 1189 - get app details

### PR DESCRIPTION
**Pull request by Phil91**
Implementation of [CPLP-1189](https://jira.catena-x.net/browse/CPLP-1189)

### CODE REVIEW

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs [hunk 1]
```diff
@@ -117,7 +118,7 @@ public async Task<(string appName, string providerName, string providerContactEm
             });
 
     /// <inheritdoc />
-    public async Task<AppDetailsData> GetAppDetailsByIdAsync(Guid appId, string? iamUserId, string? languageShortName)
+    public async Task<AppDetailsData> GetAppDetailsByIdAsync(Guid appId, string iamUserId, string? languageShortName)
```
Comment (ntruchsess) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs:121]: please check line 58-74 (as you did not modify within this PR I cannot add comments there:
        if(new []{ appDetails.Name, appDetails.Provider, appDetails.ContactEmail }.Any(d => d is null))
        {
            var nullProperties = new List<string>();
            if (appDetails.Name is null)
            {
                nullProperties.Add($"{nameof(App)}.{nameof(appDetails.Name)}");
            }
            if (appDetails.Provider is null)
            {
                nullProperties.Add($"{nameof(App)}.{nameof(appDetails.Provider)}");
            }
@@ -97,27 +98,27 @@
                UseCaseNames = a.UseCases.Select(uc => uc.Name),
                a.ThumbnailUrl,
                ShortDescription =
                    _context.Languages.SingleOrDefault(l => l.ShortName == languageShortName) == null 
                        ? null 
                        : a.AppDescriptions.SingleOrDefault(d => d.LanguageShortName == languageShortName)!.DescriptionShort
                          ?? a.AppDescriptions.SingleOrDefault(d => d.LanguageShortName == Constants.DefaultLanguage)!.DescriptionShort,
                LicenseText = a.AppLicenses
                    .Select(license => license.Licensetext)
                    .FirstOrDefault()
            }).AsAsyncEnumerable().Select(app => new AppData(
                app.Name ?? Constants.ErrorString,
                app.ShortDescription ?? Constants.ErrorString,
                app.VendorCompanyName ?? Constants.ErrorString,
                app.LicenseText ?? Constants.ErrorString,
                app.ThumbnailUrl ?? Constants.ErrorString
                )
            {
                Id = app.Id,
                UseCases = app.UseCaseNames.Select(name => name).ToList()
            });

Please move this check to the businesslogic.

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs [hunk 2]
```diff
@@ -141,9 +142,11 @@ public async Task<AppDetailsData> GetAppDetailsByIdAsync(Guid appId, string? iam
                     .Select(license => license.Licensetext)
                     .FirstOrDefault(),
                 Tags = a.Tags.Select(t => t.Name),
-                IsPurchased = iamUserId == null ?
-                    (bool?)null :
-                    a.Companies.Any(c => c.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId)),
+                IsPurchased = a.Companies.FirstOrDefault(c => c.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId)) == null ?
```
Comment (ntruchsess) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs:145]: IsPurchased = a.Companies.Where(c => c.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId)).SelectMany(company => company.CompanyAssignedApps.Where(x => x.AppId == appId)).Select(x => x.AppSubscriptionStatusId).FirstOrDefault(),

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs (hunk 1 above)
Comment (Phil91) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs:98]: moved it

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs (hunk 2 above)
Comment (Phil91) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppRepository.cs]: thanks. if changed the query

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**